app/web/features/profile/view/leaveReference/ReferenceForm.tsx

```diff
@@ -85,36 +89,34 @@ export default function ReferenceForm() {
     }));
   };
 
-  const friendMatch = useRouteMatch<ReferenceRouteParams>({
-    path: `${leaveReferenceBaseRoute}/:referenceType/:userId/:step?`,
-  });
-  const hostingMatch = useRouteMatch<ReferenceRouteParams>({
-    path: `${leaveReferenceBaseRoute}/:referenceType/:userId/:hostRequestId/:step?`,
-  });
-
-  const step =
-    friendMatch?.params.referenceType ===
-    referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND]
-      ? friendMatch?.params.step
-      : hostingMatch?.params.step;
-
   return step === undefined ? (
```

Comment:
Given the pattern I've seen your refactor so far, I think this is kinda gross, as it's mixing routing concerns within the "page" component itself which should really... be responsible for the page itself only.

Maybe make an issue about this as a low priority refactor? It's a big change here already so I think it's fine to leave it here and do separately if we decide to later.

Reply:
Good idea, I think especially now there is a nice place to keep all routing logic.